### DISCUSSION TITLE
ffmpeg priority / CPU usage settings?

### DISCUSSION THREAD

**Discussion by philstopford** [CineEncoder/cine-encoder#52] [Ideas]
During a conversion, it can be that the system starts to lag due to the heavy CPU utilization and the runtime priority of ffmpeg. It would be helpful to have some options here to control these (e.g. limiting the number of threads in use by ffmpeg, and/or controlling the relative priority of the ffmpeg process).

ffmpeg also does not seem to scale linearly with the number of threads on the host system, so it would perhaps also be useful to have the ability to run mutiple tasks at the same time, each with a limited number of threads (e.g. 2 tasks for 4 threads each).

What do you think?

**Comment by CineEncoder** [comment]
Hi!
I need to think, I don't know how to do it yet.

**Comment by CineEncoder** [comment]
Although there is a -threads parameter in ffmpeg, I'll see how it works.

**Comment by philstopford** [comment]
I have a branch here (https://github.com/philstopford/cine-encoder/tree/Threads) and sent a PR that implements this
